src/test/scala/org/marvin/executor/api/GenericHttpAPITest.scala

```diff
@@ -398,11 +404,19 @@ class GenericHttpAPITest extends WordSpec with ScalatestRouteTest with Matchers
     GenericHttpAPI.system = system
     GenericHttpAPI.onlineActionTimeout = timeout
     GenericHttpAPI.healthCheckTimeout = timeout
+
     probe
   }
+
+  def mockProtocolService(): Unit = {
+    val protocolService = mock[ProtocolUtil]
+    (protocolService.generateProtocol _).expects(*).returning("mockedProtocol")
+    GenericHttpAPI.api = new GenericHttpAPIImpl()
+    GenericHttpAPI.protocolService = protocolService
+  }
 }
 
-class GenericHttpAPIOpen extends GenericHttpAPI {
+class GenericHttpAPIOpen(var protocolService: ProtocolUtil) extends GenericHttpAPI {
```

Comment:
The test class GenericHttpAPIOpen must be analogue of the production implementation GenericHttpAPI. In this case you should modify this class here to stop using ProtocolUtil as a constructor parameter and use it as a global static variable in the GenericHttpAPI singleton object.